artichoke-backend/src/extn/core/string/mruby.rs

```diff
@@ -439,6 +441,21 @@ unsafe extern "C" fn string_empty(mrb: *mut sys::mrb_state, slf: sys::mrb_value)
     }
 }
 
+unsafe extern "C" fn string_end_with(mrb: *mut sys::mrb_state, slf: sys::mrb_value) -> sys::mrb_value {
+    let prefixes = mrb_get_args!(mrb, *args);
+    unwrap_interpreter!(mrb, to => guard);
+    let value = Value::from(slf);
+
+    for prefix in prefixes.iter().map(|&other| Value::from(other)) {
+        match trampoline::end_with(&mut guard, value, prefix) {
+            Ok(true) => return guard.convert(true).inner(),
+            Ok(false) => {}
+            Err(exception) => error::raise(guard, exception),
+        }
+    }
+    guard.convert(false).inner()
```

Comment:
I'd prefer if we moved the loop into the trampoline. You can make the trampoline take a generic iterator. Can you refactor this to use the same pattern as `Kernel#puts`?


https://github.com/artichoke/artichoke/blob/dac4c8c9280a00350ae2421628b6f1b1710171b7/artichoke-backend/src/extn/core/kernel/trampoline.rs#L129-L132

https://github.com/artichoke/artichoke/blob/dac4c8c9280a00350ae2421628b6f1b1710171b7/artichoke-backend/src/extn/core/kernel/mruby.rs#L93-L98